### PR TITLE
parse_datetime fix relative month dates result in wrong dates

### PR DESCRIPTION
**Pull request by cerdelen**
fixed: #252

Using "X months ago" or "X months" (into the future) resulted in wrong dates if the months had different amount of days. Previously parse_datetime took the days of the current month and multiplied it times x to get the day delta between now and the desired X months ago. If this month had 31 days but the previous month had only 30 days this would result in -1 month and -1 day.

New implementation loops x times and checks the days in the months to calculate the correct day delta.

### CODE REVIEW

**Comment by cerdelen** [comment]
Are these tests sufficient? @sylvestre

File: tests/time.rs [hunk 1]
```diff
@@ -129,6 +129,50 @@ fn test_time_invalid(#[case] input: &str) {
     );
 }
 
+#[rstest]
+#[case::months_ago_1("2026-01-12 1 Months ago", "2025-12-12")]
+#[case::months_ago_2("2026-01-12 2 Months ago", "2025-11-12")]
```
[the remark this answers is not in the document]
Comment (sylvestre) [reply tests/time.rs]: consider adding edge cases like 0, 24, 36 months

and very big values and negative

**Comment by cerdelen** [comment]
@sylvestre hope its okay to ping again but is there anything to be added here?

File: src/items/builder.rs [hunk 2]
```diff
@@ -1,7 +1,7 @@
 // For the full copyright and license information, please view the LICENSE
 // file that was distributed with this source code.
 
-use jiff::{civil, Span, Zoned};
+use jiff::{civil, Span, ToSpan, Zoned};
```
Comment (sylvestre) [inline src/items/builder.rs]: do we really use ToSpan ?

File: src/items/builder.rs [hunk 3]
```diff
@@ -276,13 +276,21 @@ impl DateTimeBuilder {
...
+            dt = match rel {
+                relative::Relative::Years(_) | relative::Relative::Months(_) => {
+                    // GNU way of calculating relative months and years
+                    // GNU changes the month and then checks if the target month has
+                    // this day. If this day does not exist in the target month it overflows
+                    // the difference
+                    let desired_day = dt.day();
+                    dt = dt.checked_add::<Span>(rel.try_into()?)?;
+                    if desired_day != dt.day() {
+                        dt = dt.checked_add((desired_day - dt.day()).days())?;
```
Comment (sylvestre) [inline src/items/builder.rs]: maybe needs checked arithmetic 
Potential integer underflow in subtraction (desired_day - dt.day())  ?

File: src/items/builder.rs (hunk 2 above)
Comment (cerdelen) [reply src/items/builder.rs]: removing it gives me an error on line 288. the last .day() invocation uses it apparently.

**Comment by ChrisDryden** [comment]
Surprisingly this also is the cause of the `tests/ls/abmon-align` gnu tests since it uses touch to change the dates for the ls test
